torch/ao/pruning/_experimental/pruner/lstm_saliency_pruner.py

```diff
@@ -0,0 +1,34 @@
+from typing import cast
+
+import torch
+from .base_structured_sparsifier import BaseStructuredSparsifier, FakeStructuredSparsity
+
+class LSTMSaliencyPruner(BaseStructuredSparsifier):
+    """
+    Prune packed LSTM weights based on saliency
+    """
+
+    def update_mask(self, module, tensor_name, **kwargs):
+        weights = getattr(module, tensor_name)
+
+        for p in getattr(module.parametrizations, tensor_name):
+            if isinstance(p, FakeStructuredSparsity):
+                mask = cast(torch.Tensor, p.mask)
+
+                # select weights based on magnitude
+                if weights.dim() <= 1:
+                    raise Exception("Structured pruning can only be applied to a 2+dim weight tensor!")
+                # take norm over all but first dim
+                dims = tuple(range(1, weights.dim()))
+                saliency = weights.norm(dim=dims, p=1)
+
+                # handle weights in 4 groups
+                split_size = len(mask) // 4
```

Comment:
why do we handle weights in 4 groups?

Reply:
There are four sets of weights in each tensor. It may be worth verifying whether they should be pruned completely independently.
https://pytorch.org/docs/stable/generated/torch.nn.LSTM.html#:~:text=of%20shape%20(-,4*hidden_size,-%2C%20input_size)%20for
(search for "4*hidden_size" if the link doesn't jump for you)

Reply:
I see, thanks, would be helpful to add a comment I think